### ISSUE TITLE
Add support for Suspend and Resume

### ISSUE THREAD

**Issue by erwin** [quickemu-project/quickemu#291]
`virsh` has commands like `suspend` `dompmsuspend` `managedsave`

`qemu` has `savevm` and `loadvm`, but I think these require the `-qmp` switch when starting qemu.

Has anyone using `quickemu` made a patch to enable savevm / loadvm or some other suspend/resume function?

Looks like this documentation could get us part of the way there:

https://wiki.qemu.org/Documentation/QMP

Since it's more trouble than it's worth to integrate `quickemu` with libvirt, would sure be nice if we could still build some very simple `suspend` / `resume` support into `quickemu`.

---

UPDATE: 

To use the `qemu` monitor function, we need a way to connect to it. Could be telnet or a local domain socket. To use a local domain socket, in `./quickemu` change:

```diff
-         -monitor none
+         -monitor unix:${VMDIR}/monitor,server,nowait
```

Then you can connect to the monitor with:

```
cd (to your vm directory so the monitor socket is found)
socat -,echo=0,icanon=0 unix-connect:monitor
```

However, I haven't been able to get `savevm` to work on Windows or Mac VMs.
The `monitor` does allow to send a system powerdown from the command line, which is helpful, but I really want suspend and resume so that I can integrate it into my build process, without having to keep the whole VM wasting CPU and memory 100% of the time.

**Comment by ChrisMacGregor** [comment]
Did you ever get savevm to work (for Windows or macOS)?  I would like to have the same thing you describe, for the same reason.

**Comment by erwin** [comment]
Sorry, no luck yet on savevm.
For now I just run the both of the vms headless in the background, connect, compile and unit test via ssh... If the desktop UI isn't being refreshed it takes the load down some, though I would definitely love to suspend these between compiles, and allocate a ton of resources to them temporarily while building/testing. 

I'm also sharing files to each vm via samba, which has been more reliable for me than sftp/sshfs

**Comment by flexiondotorg** [comment]
Fixed via #347
